onnxruntime/test/onnx/runner.cc

```diff
@@ -489,30 +511,30 @@ void RunSingleTestCase(const ITestCase& info, Ort::Env& env, const Ort::SessionO
       concurrent_runs = 1;
     }
 
+    DataRunner* r;
     if (concurrent_runs > 1 && data_count > 1) {
-      r.reset(new PTestRunner(session_object.release(), info, tpool, on_finished));
+      r = new PTestRunner(session_object.release(), info, tpool, on_finished);
     } else {
-      r.reset(new SeqTestRunner(session_object.release(), info, repeat_count, on_finished));
+      r = new SeqTestRunner(session_object.release(), info, repeat_count, on_finished);
     }
     r->Start(pci, concurrent_runs);
 
-    // DataRunner::Finish will delete itself, so now that we know everything has started without any exceptions
-    // we can release it from the unique_ptr
-    r.release();
+    // both PTestRunner and SeqTestRunner call DataRunner::Finish which will delete itself and call on_finished callback
+    // at this point we know everything has strated without any exceptions so simply return.
     return;
-  } catch (const Ort::Exception& ex) {
-    if (ex.GetOrtErrorCode() != ORT_NOT_IMPLEMENTED) {
-      throw;
-    }
 
+  } catch (const onnxruntime::NotImplementedException& ex) {
     LOGF_DEFAULT(ERROR, "Test %s failed:%s", info.GetTestCaseName().c_str(), ex.what());
     std::string node_name;
     ret = std::make_shared<TestCaseResult>(data_count, EXECUTE_RESULT::NOT_SUPPORT, "");
-  } catch (onnxruntime::NotImplementedException& ex) {
+  } catch (const Ort::Exception& ex) {
     LOGF_DEFAULT(ERROR, "Test %s failed:%s", info.GetTestCaseName().c_str(), ex.what());
```

Comment:
>, "Test %s failed:%s", inf [](start = 22, length = 26)

Here it does not represent test failure

Reply:
It does... If session creation fails then it should be treated as a test failure no?

Reply:
This is separate. Failed to load model file most likely.

---
In reply to: [452998280](https://github.com/microsoft/onnxruntime/pull/4475#discussion_r452998280) [](ancestors = 452998280)